### PR TITLE
Add Rounded Backgrounds to Widgets for Consistent UI Across APIs

### LINKED ISSUE #17656: Inconsistent Widget Styling
<p height="200">
  
  
</p>

Currently, the first widget uses dynamic system colors and lacks corner rounding. The second widget has corner rounding, adapts to light/dark mode, and has a more polished design.

## Expected Behavior:
The first widget should be updated to match the second widget's design:
- Add corner rounding to the first widget.
- Ensure the first widget adapts to light/dark mode, just like the second widget.

This will ensure consistency in design across both widgets.

### PR DESCRIPTION
**Pull request by xenonnn4w**
## Purpose / Description
The current widget layouts vary across devices, these changes will ensure consistency across all the devices.

## Fixes
* Fixes #17656

## Approach
https://github.com/ankidroid/Anki-Android/issues/17656#issuecomment-2561297107

## How Has This Been Tested?
Motorola-G04-API34
Redmi Note 7 Pro - API29
Pixel 5 (emulator) - API28
Pixel 5 (emulator) - API35
One Plus Nord Ce - API34
Samsung A34 - API32

## Results 
| Device       | Result                                                                 |
|-------------------------|----------------------------------------------------------------------|
| Samsung |  |
| Motorola Device         | Motorola Device |
| Emulator        | Pixel 5 |
| Redmi Note      |  |
| One Plus      |  |

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by lukstbit** [comment]
Did you also check on the minimum API level we support, 24?

**Comment by xenonnn4w** [comment]
Yes working as expected.

**Review by david-allison — APPROVED** [review]
Cheers!

File: AnkiDroid/src/main/res/drawable/widget_deck_picker_rounded_background.xml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<shape xmlns:android="http://schemas.android.com/apk/res/android" android:shape="rectangle">
+    <corners android:radius="28dp" />
+    <solid android:color="?attr/colorSurface" />
```
Comment (lukstbit) [inline AnkiDroid/src/main/res/drawable/widget_deck_picker_rounded_background.xml:3]: Shouldn't this be one of the _container*_ colors(most likely _?attr/colorPrimaryContainer_)? Right now the two widgets put side by side look pretty different although they come from the same app.

**Review by lukstbit — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/res/drawable/widget_deck_picker_rounded_background.xml (hunk 1 above)
Comment (xenonnn4w) [reply AnkiDroid/src/main/res/drawable/widget_deck_picker_rounded_background.xml:3]: Great observation! My intention was to keep the deck picker minimal and elegant, using a single subtle color to maintain simplicity. On the other hand, the card analysis has two colors, which necessitates a touch of variation in the color shades to effectively implement that design.

**Comment by xenonnn4w** [comment]
> although they come from the same app.

This seems to be a valid point, after i played some variations of colors. Thanks

**Review by lukstbit — APPROVED** [review]
Thanks, I do think being consistent is nice.

**Pull request merged**